### DISCUSSION TITLE
How to convert/extract Fastify request to Node.js core request?

### DISCUSSION THREAD

**Discussion by lilouartz** [fastify/fastify#5610] [General]
Node.js has a global type Request. It seems to be coming from undici:

```ts
interface Request extends _Request {}
var Request: typeof globalThis extends {
    onmessage: any;
    Request: infer T;
} ? T
    : typeof import("undici-types").Request;
```

How do I convert Fastify request to this request type?

**Comment by metcoder95** [comment]
The request type are different. The one coming from Undici references the WHATWG `Request` object, meanwhile the one used by Fastify comes from [`http.ClientRequest`](https://nodejs.org/docs/latest/api/http.html#class-httpclientrequest) (with extensions added to build `FastifyRequest`)

**Reply by lilouartz** [reply]
I understand that they are different. How do I convert from one to another?

**Reply by metcoder95** [reply]
You cannot, they do not share overlapping properties as they serve different purpose.

Not sure what are you trying to achieve, but one option is to decorate the `FastifyRequest` with an instance of the WHATWG `Request`

**Reply by Uzlopak** [reply]
Well it is possible to a degree. Just instantiate the Request Object with fields from the FastifyRequest Instace. 

You will probably not find an existing solution. So just write it yourself.
